slick-codegen/src/main/scala/slick/codegen/AbstractSourceCodeGenerator.scala

```diff
@@ -57,14 +57,42 @@ abstract class AbstractSourceCodeGenerator(model: m.Model)
       else compoundValue(types)
     }
 
-    def compoundValue(values: Seq[String]): String = {
+    //build and nest tuples if > 22 elements
+    def tupleBuilder(values: Seq[String] , prefix: String): String = {
+      def mkTuple[T](l: Seq[T], _prefix: String) = l.map{ _prefix + _ }.mkString("(" , " ," , ")")
+      if(values.size <= 22) 
+        mkTuple(values,"") 
+      else 
+        mkTuple( values.grouped(22).toList.map {mkTuple(_,prefix)} , ""  )
+    }
+
+    def tupleFactoryBuilder(columns: Seq[Column]): String = {
+        if(columns.size <= 22 ) s"${TableClass.elementType}.tupled"
+        else{
+          val count = (columns.size / 22.0).ceil.toInt
+          val tuples = Range(0,count).map{ "t" + _ }
```

Comment:
so tupleBuilder now works with a table > 440 columns. How about tupleFactoryBuilder? the above two lines do not look like it.